server/routes/summary_routes.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter } from '../../../../src/core/server';
+import { SUMMARY_ASSISTANT_API } from '../../common/constants/llm';
+import { getOpenSearchClientTransport } from '../utils/get_opensearch_client_transport';
+import { getAgent, searchAgentByName } from './get_agent';
+import { AssistantServiceSetup } from '../services/assistant_service';
+
+const SUMMARY_AGENT_CONFIG_ID = 'os_summary';
+const OS_INSIGHT_AGENT_CONFIG_ID = 'os_insight';
+let osInsightAgentId: string | undefined;
+let userInsightAgentId: string | undefined;
+
+export function registerSummaryAssistantRoutes(
+  router: IRouter,
+  assistantService: AssistantServiceSetup
+) {
+  router.post(
+    {
+      path: SUMMARY_ASSISTANT_API.SUMMARIZE,
+      validate: {
+        body: schema.object({
+          type: schema.string(),
+          insightType: schema.maybe(schema.string()),
+          question: schema.string(),
+          context: schema.maybe(schema.string()),
+        }),
+        query: schema.object({
+          dataSourceId: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    router.handleLegacyErrors(async (context, req, res) => {
+      const client = await getOpenSearchClientTransport({
+        context,
+        dataSourceId: req.query.dataSourceId,
+      });
+      const assistantClient = assistantService.getScopedClient(req, context);
+      const response = await assistantClient.executeAgentByName(SUMMARY_AGENT_CONFIG_ID, {
+        context: req.body.context,
+        question: req.body.question,
+      });
+      let summary;
+      let insightAgentIdExists = false;
+      try {
+        if (req.body.insightType) {
+          // We have separate agent for os_insight and user_insight. And for user_insight, we can
+          // only get it by searching on name since it is not stored in agent config.
+          if (req.body.insightType === 'os_insight') {
+            if (!osInsightAgentId) {
+              osInsightAgentId = await getAgent(OS_INSIGHT_AGENT_CONFIG_ID, client);
+            }
+            insightAgentIdExists = !!osInsightAgentId;
+          } else if (req.body.insightType === 'user_insight') {
+            if (req.body.type === 'alerts') {
+              if (!userInsightAgentId) {
+                userInsightAgentId = await searchAgentByName('KB_For_Alert_Insight', client);
+              }
+            }
+            insightAgentIdExists = !!userInsightAgentId;
+          }
+        }
```

Comment:
nit: we can refactor these two api calls run in parallel with Promise.all() as they don't depend on each other